lib/routes.js

```diff
@@ -352,13 +357,18 @@ module.exports = async function (app, opts) {
 
     if (allowBatchedQueries && Array.isArray(request.body)) {
       // Batched query
-      return Promise.all(request.body.map(r =>
-        execute(r, request, reply)
-          .catch(e => {
-            const { response } = errorFormatter(e, request[kRequestContext])
-            return response
-          })
-      ))
+      Object.assign(request[kRequestContext], { operationsCount: request.body.length })
+
+      return Promise.all(request.body.map(async (r, operationId) => {
+        Object.assign(request[kRequestContext], { operationId })
```

Comment:
⚠️ this is a race condition

Reply:
I don't think so. Promises are synchronous until reaching an async io operation. In this case `execute()`.

Reply:
I don't think `request[kRequestContext]` will ever be null or undefined at that point, it gets created [right here](https://github.com/drakhart/mercurius/blob/c31bee092d344e458c3f6621147d2b2b5cf1b389/lib/routes.js#L348) before executing the query operation a few lines below.

Reply:
And yes, that's certainly a race condition due to the `Promise.all`. Looking into it.

Reply:
Race condition should be solved now, by cloning the whole request object (not just Mercurius' context inside it) instead of mutating the original one inside the `Promise.all`.